llvm/lib/Object/RecordStreamer.h

```diff
@@ -28,11 +29,11 @@ class RecordStreamer : public MCStreamer {
 
 private:
   const Module &M;
-  StringMap<State> Symbols;
+  MapVector<StringRef, State> Symbols;
```

Comment:
I would not expect nondeterminism from Symbols
Order in hash table should be determined character in StringRefs and order they inserted.
Should be the same all the time?

Reply:
Symbols generation order comes from traversing SymverAliasMap in flushSymverDirectives func, which is an unordered map data structure, and I understand that traversing creates nondeterminism problems.

Reply:
unordered does not mean nondeterminism.

So I believe change in this line is unnecessary.

Reply:
To clarify, order is determined by:
1. order and values passed into markDefined/markDefined/markUsed invoked
2. hash table implementation

So for the same lld binary those are consistent, and should not change result.

SymverAliasMap on other hand, also depends on raw pointers values, which controlled by memory allocator, which are very inconsistent, especially with ASLR.

So could you please confirm that `SymverAliasMap` was the root-cause of your issue, and not `Symbols`?

Reply:
Ok, thanks for your reply, Symbols may not need to change StringMap to MapVec. I'll check again if modifying only SymverAliasMap is OK.

Reply:
I've made a confirmation that SymverAliasMap was the root-cause of your issue, and not Symbols.

Reply:
Thank you!